app/ent/user.go

```diff
@@ -0,0 +1,17 @@
+// Package ent will consist of key entities  of our project
+// and represents core/domain layer of our app -  correct me if I am wrong
+// and would not have any dependencies on other layers.
+// ent package also contains validations and custom errors.
+package ent
+
+// User is key entity in our project
+// Entities like User are the least likely to change
+// when something external changes.
+type User struct{}
+
+// TODO: sentinel errors
+
+// Validate will validate User's signup data
+func (u *User) Validate() error {
```

Comment:
I see no reason why we need pointer to user here.
We add it when it will be needed.

Reply:
I see a reason to write some meth-related to User that would change its state. In that case, we have to follow same signature with * in all meths. If you see no reason for now let's remove all pointers.